src/librustc_codegen_ssa/base.rs

```diff
@@ -200,6 +231,81 @@ pub fn unsize_thin_ptr<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
     }
 }
 
+/// Coerces `op` to `dst.ty`. `op` may be a fat pointer.
+pub fn coerce_ptr_unsized<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
```

Comment:
Something I should've said a while back: `base` and `common` *were always* organizational disasters, we should be moving abstractions closer to where they're used (e.g. `mir::rvalue` in this case, and it would probably be a method, there).

Reply:
Fair enough, but probably something to tidy up in a follow-up PR.

Reply:
It would help this PR make sense if it happened before, IMO.